hudi-common/src/main/java/org/apache/hudi/common/util/JsonUtils.java

```diff
@@ -20,41 +20,74 @@
 package org.apache.hudi.common.util;
 
 import org.apache.hudi.exception.HoodieIOException;
+import org.apache.hudi.util.Lazy;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import com.fasterxml.jackson.annotation.PropertyAccessor;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.util.StdDateFormat;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 
 /**
  * Utils for JSON serialization and deserialization.
  */
 public class JsonUtils {
 
-  private static final ObjectMapper MAPPER = new ObjectMapper();
-
-  static {
-    MAPPER.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-    // We need to exclude custom getters, setters and creators which can use member fields
-    // to derive new fields, so that they are not included in the serialization
-    MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
-    MAPPER.setVisibility(PropertyAccessor.GETTER, JsonAutoDetect.Visibility.NONE);
-    MAPPER.setVisibility(PropertyAccessor.IS_GETTER, JsonAutoDetect.Visibility.NONE);
-    MAPPER.setVisibility(PropertyAccessor.SETTER, JsonAutoDetect.Visibility.NONE);
-    MAPPER.setVisibility(PropertyAccessor.CREATOR, JsonAutoDetect.Visibility.NONE);
-  }
+  private static final Lazy<ObjectMapper> MAPPER = Lazy.lazily(JsonUtils::instantiateObjectMapper);
 
   public static ObjectMapper getObjectMapper() {
-    return MAPPER;
+    return MAPPER.get();
   }
 
   public static String toString(Object value) {
     try {
-      return MAPPER.writeValueAsString(value);
+      return MAPPER.get().writeValueAsString(value);
     } catch (JsonProcessingException e) {
       throw new HoodieIOException(
           "Fail to convert the class: " + value.getClass().getName() + " to Json String", e);
     }
   }
+
+  private static ObjectMapper instantiateObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+
+    registerModules(mapper);
+
+    // We're writing out dates as their string representations instead of (int) timestamps
+    mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+    // NOTE: This is necessary to make sure that w/ Jackson >= 2.11 colon is not infixed
+    //       into the timezone value ("+00:00" as opposed to "+0000" before 2.11)
+    //       While Jackson is able to parse both of these formats, we keep it as false
+    //       to make sure metadata produced by Hudi stays consistent across Jackson versions
+    configureColonInTimezone(mapper);
```

Comment:
Seems the main fix is line 60 ~ line 65.
Wondering why we serialize the data type as date time string instead of epoch days. Can you clarify what is the scope of the change, because the column stats uses Avro serialization which store the data as java `LocalDate` already, curious how the change could affect column stats serialization.

If we remove line 60 ~ 65, test `TestColumnStatsIndex` still passes, so what exactly are we fixing here?

Reply:
I think L57 is the essential part for this fix. It registers `JavaTimeModule` so jackson can handle `LocalDate`, which is not supported by default

Reply:
Yeah, wondering when we need to serialize the column stats to json string?

Reply:
I think we serialize the column stats to the metadata record payload, correct?